src/Symfony/Bundle/SecurityBundle/DataCollector/SecurityDataCollector.php

```diff
@@ -104,6 +108,20 @@ public function collect(Request $request, Response $response, \Exception $except
                 'supports_role_hierarchy' => null !== $this->roleHierarchy,
             );
         }
+
+        // collect voters and access decision manager information
+        if (null !== $this->accessDecisionManager) {
+            $this->data['access_decision_log'] = $this->accessDecisionManager->getDecisionLog();
+            $this->data['voter_strategy'] = $this->accessDecisionManager->getStrategy();
```

Comment:
Could typehint against the `DebugAccessDecisionManager`, good to let it crash if you run this incorrectly imo